azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/DeployMojo.java

```diff
@@ -89,65 +90,111 @@ protected void doExecute() throws MojoExecutionException, AzureExecutionExceptio
+            getLog().info(toCreateDeployment ? CREATE_DEPLOYMENT_DOING : UPDATE_DEPLOYMENT_DOING);
+            deploymentModifier.commit();
+            getLog().info(toCreateDeployment ? CREATE_DEPLOYMENT_DONE : UPDATE_DEPLOYMENT_DONE);
+        }));
+        if (!appUpdater.isSkippable()) {
+            tasks.add(new AzureTask<Void>(UPDATE_APP_TITLE, () -> {
+                getLog().info(UPDATE_APP_DOING);
+                appUpdater.commit();
+                getLog().info(UPDATE_APP_DONE);
+                getLog().warn(UPDATE_APP_WARNING);
+            }));
+        }
+
+        tasks.add(new AzureTask<Void>(() -> {
+            if (!noWait) {
+                getLog().info(GET_DEPLOYMENT_STATUS_DOING);
+                if (!deployment.waitUntilReady(GET_STATUS_TIMEOUT)) {
+                    getLog().warn(GET_DEPLOYMENT_STATUS_TIMEOUT);
+                }
+            }
+            printStatus(deployment);
+            printPublicUrl(app);
+        }));
+        if (this.confirmDeploy(tasks)) {
+            tasks.forEach(AzureTask::go);
+        }
+    }
 
-        if (!noWait) {
-            deployment.waitUntilReady(GET_STATUS_TIMEOUT);
+    protected boolean confirmDeploy(List<AzureTask<?>> tasks) throws MojoFailureException {
+        try (IPrompter prompter = new DefaultPrompter()) {
+            getLog().info(CONFIRM_PROMPT_START);
+            tasks.stream().filter(t -> StringUtils.isNotEmpty(t.getTitle())).forEach((t) -> System.out.printf("\t*  %s%n", t.getTitle()));
+            return prompter.promoteYesNo(CONFIRM_PROMPT_CONFIRM, true, true);
+        } catch (IOException e) {
+            throw new MojoFailureException(e.getMessage());
         }
-        printStatus(deployment);
-        printPublicUrl(app);
     }
 
     protected void printPublicUrl(SpringCloudApp app) {
         if (!app.entity().isPublic()) {
             return;
         }
-        getLog().info(GETTING_PUBLIC_URL);
+        getLog().info(GET_APP_URL_DOING);
         String publicUrl = app.entity().getApplicationUrl();
         if (!noWait && StringUtils.isEmpty(publicUrl)) {
             publicUrl = RxUtils.pollUntil(() -> app.refresh().entity().getApplicationUrl(), StringUtils::isNoneEmpty, GET_URL_TIMEOUT);
         }
         if (StringUtils.isEmpty(publicUrl)) {
-            getLog().warn(GET_APP_URL_FAIL);
+            getLog().warn(GET_APP_URL_FAILURE);
         } else {
-            getLog().info(String.format(GET_APP_URL_SUCCESSFULLY, publicUrl));
+            getLog().info(String.format(GET_APP_URL_SUCCESS, publicUrl));
         }
     }
 
     protected void printStatus(SpringCloudDeployment deployment) {
         if (!AzureSpringCloudConfigUtils.isDeploymentDone(deployment)) {
             getLog().warn(GET_DEPLOYMENT_STATUS_TIMEOUT);
         }
-        System.out.printf("Deployment Status: %s%n", deployment.entity().getStatus());
+        getLog().info(String.format("Deployment Status: %s", deployment.entity().getStatus()));
```

Comment:
Logger should handle place holder. No need String.format() here / and other places.

getLog().info("Deployment Status: {}", status)

Reply:
no, it does not support template.